pygeos/_geom.pyx

```diff
@@ -0,0 +1,82 @@
+from cpython cimport PyObject
+cimport cython
+
+import numpy as np
+cimport numpy as np
+import pygeos
+
+from pygeos._geos cimport (
+    GEOSContextHandle_t,
+    GEOSGeometry,
+    GEOSGeom_clone_r,
+    GEOSGetGeometryN_r,
+    GEOS_init_r,
+    GEOS_finish_r,
+    get_geos_handle
+)
+from pygeos._pygeos_api cimport (
+    import_pygeos_c_api,
+    PyGEOS_CreateGeometry,
+    PyGEOS_GetGEOSGeometry
+)
+
+# initialize PyGEOS C API
+import_pygeos_c_api()
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def get_parts(object[:] array):
+    cdef Py_ssize_t geom_idx = 0
+    cdef Py_ssize_t part_idx = 0
+    cdef Py_ssize_t idx = 0
+    cdef GEOSGeometry *geom = NULL
+    cdef GEOSGeometry *part = NULL
+
+    cdef const np.intp_t [:] input_index_view = np.arange(0, len(array), dtype=np.intp)
+
+    counts = pygeos.get_num_geometries(array)
+    cdef int [:] counts_view = counts[:]
```

Comment:
```suggestion
    cdef int[:] counts_view = counts[:]
```

(small comment for consistency in style)

In addition, I am not sure the final `[:]` in ``..= counts[:]`` is actually needed (I see this pattern in pandas as well where we don't use the extra `[:]`)